src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -84,19 +84,26 @@ public void lockExclusive(UUID datasetId, String flightId) {
         MapSqlParameterSource params = new MapSqlParameterSource()
             .addValue("datasetid", datasetId)
             .addValue("flightid", flightId);
-        int numRowsUpdated = jdbcTemplate.update(sql, params);
 
-        // if no rows were updated, then throw an exception
-        if (numRowsUpdated == 0) {
-            // this method checks if the dataset exists
-            // if it does not exist, then the method throws a DatasetNotFoundException
-            // we don't need the result (dataset summary) here, just the existence check, so ignore the return value.
-            retrieveSummaryById(datasetId);
+        // fault insert for tests DatasetConnectedTest > retryAndAcquireExclusiveLock
+        // & retryAndFailAcquireExclusiveLock
+        DataAccessException faultToInsert = getExclusiveFaultToInsert();
+        runUpdateAndInsertFault(sql, params, faultToInsert, true, datasetId);
 
-            // otherwise, throw a lock exception
-            logger.debug("numRowsUpdated=" + numRowsUpdated);
-            throw new DatasetLockException("Failed to lock the dataset");
+        logger.debug("Exclusive lock acquired for dataset {}, flight {}", datasetId, flightId);
+    }
+
+    private DataAccessException getExclusiveFaultToInsert() {
```

Comment:
```suggestion
    private DataAccessException getExclusiveLockFaultToInsert() {
```
Perhaps switch the name to be consistent with section below or refactor into a single method?

Reply:
OK - I've refactored into a single method.